### PR TITLE
[MB-1161]Add secure vault support to the andes client

### PR DESCRIPTION
**Pull request by prabathariyaratna**

### CODE REVIEW

**Comment by prabathariyaratna** [comment]
This addressed the jira https://wso2.org/jira/browse/MB-1161

File: modules/andes-core/client/src/main/java/org/wso2/andes/jndi/PropertiesFileInitialContextFactory.java [hunk 1]
```diff
@@ -164,6 +166,8 @@ protected ReadOnlyContext createContext(Map data, Hashtable environment)
 
     protected void createConnectionFactories(Map data, Hashtable environment) throws ConfigurationException
     {
+        resolveEncryptedProperties(environment);
```
Comment (sajinidesilva) [inline modules/andes-core/client/src/main/java/org/wso2/andes/jndi/PropertiesFileInitialContextFactory.java]: Shall we add a comment to this method even though you didn't create this method?

File: modules/andes-core/client/src/main/java/org/wso2/andes/jndi/PropertiesFileInitialContextFactory.java [hunk 2]
```diff
@@ -180,6 +184,45 @@ protected void createConnectionFactories(Map data, Hashtable environment) throws
...
+            SecretResolver secretResolver = SecretResolverFactory.create(properties);
+
+            for (Object key : environment.keySet()) {
+
+                if (secretResolver != null && secretResolver.isInitialized()) {
+                    if (secretResolver.isTokenProtected(key.toString())) {
+                        environment.put(key.toString(), secretResolver.resolve(key.toString()));
+                    }
+                }
+
```
Comment (shafreenAnfar) [inline modules/andes-core/client/src/main/java/org/wso2/andes/jndi/PropertiesFileInitialContextFactory.java]: Remove the space

File: modules/andes-core/client/src/main/java/org/wso2/andes/jndi/PropertiesFileInitialContextFactory.java [hunk 3]
```diff
@@ -180,6 +184,45 @@ protected void createConnectionFactories(Map data, Hashtable environment) throws
...
+            }
+
```
Comment (shafreenAnfar) [inline modules/andes-core/client/src/main/java/org/wso2/andes/jndi/PropertiesFileInitialContextFactory.java]: Remove the space.

Comment (prabathariyaratna) [inline modules/andes-core/client/src/main/java/org/wso2/andes/jndi/PropertiesFileInitialContextFactory.java]: Fixed in https://github.com/wso2/andes/pull/408

File: modules/andes-core/client/src/main/java/org/wso2/andes/jndi/PropertiesFileInitialContextFactory.java (hunk 1 above)
Comment (prabathariyaratna) [inline modules/andes-core/client/src/main/java/org/wso2/andes/jndi/PropertiesFileInitialContextFactory.java]: Fixed in https://github.com/wso2/andes/pull/408

File: modules/andes-core/client/src/main/java/org/wso2/andes/jndi/PropertiesFileInitialContextFactory.java (hunk 2 above)
Comment (prabathariyaratna) [inline modules/andes-core/client/src/main/java/org/wso2/andes/jndi/PropertiesFileInitialContextFactory.java]: Fixed in https://github.com/wso2/andes/pull/408

**Pull request merged**